src/test/java/uk/gov/hmcts/reform/logging/layout/ReformLoggingLayoutTest.java

```diff
@@ -0,0 +1,139 @@
+
+    private class DummyP2Exception extends AbstractLoggingException {
+        DummyP2Exception(String message) {
+            super(AlertLevel.P2, message);
+        }
+    }
+
+    private class DummyP3Exception extends AbstractLoggingException {
+        DummyP3Exception(String message) {
+            super(AlertLevel.P3, message);
+        }
+    }
+
+    private void configLogback(String config) throws JoranException, IOException {
+        LoggerContext loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
+        JoranConfigurator configurator = new JoranConfigurator();
+
+        InputStream configStream = getClass().getClassLoader().getResourceAsStream(config);
+        configurator.setContext(loggerContext);
+        configurator.doConfigure(configStream);
+        configStream.close();
+
+        // capture console
+
+        baos = new ByteArrayOutputStream();
+        PrintStream ps = new PrintStream(baos);
+        System.setOut(ps);
+    }
+
+    @After
+    public void resetConsole() {
+        System.out.flush();
+        System.setOut(old);
+    }
+
+    @Test
+    public void testDefaultOutput() throws JoranException, IOException {
+        configLogback(LOGBACK_WITH_THREAD);
+
+        log.info("message");
+
+        String timestamp = "\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}\\.\\d{3}";
+        String thread = "\\[" + Thread.currentThread().getName() + "\\] ";
+        String logger = this.getClass().getCanonicalName();
+
+        assertThat(baos.toString()).containsPattern(
+            timestamp + " INFO  " + thread + logger + ":\\d+: message\n"
+        );
+    }
+
+    @Test
+    public void testDefaultOutputWithP2Exception() throws JoranException, IOException {
+        configLogback(LOGBACK_WITH_THREAD);
+
+        log.error("message", new DummyP2Exception("oh no"));
+
+        String timestamp = "\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}\\.\\d{3}";
+        String thread = "\\[" + Thread.currentThread().getName() + "\\] ";
+        String logger = this.getClass().getCanonicalName();
+
+        assertThat(baos.toString()).containsPattern(
+            timestamp + " ERROR " + thread + logger + ":\\d+: \\[P2\\] message\n"
+        );
+    }
+
+    @Test
+    public void testDefaultOutputWithBadException() throws JoranException, IOException {
+        configLogback(LOGBACK_WITH_THREAD);
+
+        log.error("message", new InvalidClassException("oh no"));
+
+        String timestamp = "\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}\\.\\d{3}";
+        String thread = "\\[" + Thread.currentThread().getName() + "\\] ";
+        String logger = ReformLoggingLayout.class.getCanonicalName();
+        String errorClass = InvalidClassException.class.getCanonicalName();
+        String message = String.format("Bad implementation of '%s' in use", errorClass);
+
+        assertThat(baos.toString()).containsPattern(
+            timestamp + " ERROR " + thread + logger + ":\\d+: \\[P1\\] " + message + "\n"
+        );
```

Comment:
can you add some comments here?
Does the originally intended logging happen as well, or just stuff logged by `ReformLoggingLayout` it its' own logger?